scripts/benchmarking_dataset.py

```diff
@@ -0,0 +1,232 @@
+import argparse
+import random
+
+import pandas as pd
+
+from docarray import DocumentArray
+from rich.console import Console
+from rich.table import Table
+import h5py
+import numpy as np
+
+from benchmarking_utils import (
+    create,
+    read,
+    update,
+    delete,
+    find_by_condition,
+    find_by_vector,
+    get_docs,
+    fmt,
+    recall_from_numpy,
+    save_benchmark_df,
+    plot_results,
+)
+
+
+def get_configuration_storage_backends(argparse):
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--default-hnsw',
+        help='Whether to use default HNSW configurations',
+        action='store_true',
+    )
+
+    args = parser.parse_args()
+
+    if args.default_hnsw:
+        storage_backends = [
+            ('memory', None),
+            ('sqlite', None),
+            (
+                'annlite',
+                {'n_dim': D},
+            ),
+            ('qdrant', {'n_dim': D, 'scroll_batch_size': 8}),
+            ('weaviate', {'n_dim': D}),
+            ('elasticsearch', {'n_dim': D}),
+        ]
+    else:
+        storage_backends = [
+            ('memory', None),
+            ('sqlite', None),
+            (
+                'annlite',
+                {
+                    'n_dim': D,
+                    'ef_construction': 100,
+                    'ef_search': 100,
+                    'max_connection': 16,
+                },
+            ),
+            (
+                'qdrant',
+                {'n_dim': D, 'scroll_batch_size': 8, 'ef_construct': 100, 'm': 16},
+            ),
+            (
+                'weaviate',
+                {'n_dim': D, 'ef': 100, 'ef_construction': 100, 'max_connections': 16},
+            ),
+            ('elasticsearch', {'n_dim': D, 'ef_construction': 100, 'm': 16}),
+        ]
+    return storage_backends
+
+
+def run_benchmark(
+    X_tr, X_te, dataset, n_index_values, n_vector_queries, n_query, storage_backends
+):
+    table = Table(
+        title=f'DocArray Benchmarking n_index={n_index_values[-1]} n_query={n_query} D={D} K={K}'
```

Comment:
I guess since n_index_values always contain 1 element, maybe it shouldn't be a list